### PR TITLE
Add navigator as ancestor to Offset

### PR DESCRIPTION
**Pull request by ErickPariseVitorassi**
Add `navigator.context.findRenderObject()` as an ancestor to position the overlay offset. This is required to work correctly inside a RouterOutlet, the flutter overlay widgets implement this, for example DropdownButton.

### CODE REVIEW

File: lib/src/menu_float.dart [hunk 1]
```diff
@@ -81,8 +81,12 @@ class _MenuFloatState<T> extends State<MenuFloat<T>>
   /// Returns a MenuFloatIdealPosition.
   MenuFloatIdealPosition _getWidgetPositionAndSizeRelativeToWindow(
       BuildContext? menuKeyContext) {
+    final navigator = Navigator.of(context);
     final RenderBox renderBox = menuKeyContext?.findRenderObject() as RenderBox;
-    final position = renderBox.localToGlobal(Offset.zero);
+    final position = renderBox.localToGlobal(
+      Offset.zero, 
+      ancestor: navigator.context.findRenderObject(),
```
Comment (emirdeliz) [inline lib/src/menu_float.dart:88]: What's the reason for this change?

Comment (ErickPariseVitorassi) [reply lib/src/menu_float.dart:88]: This is required to work correctly inside a RouterOutlet

**Review by emirdeliz — APPROVED** [review]
